src/app/core/services/mocks/mock.issue.service.ts

```diff
@@ -0,0 +1,394 @@
+         // `## State the duplicated issue here, if any\n${issue.duplicateOf ? `Duplicate of #${issue.duplicateOf}` : `--`}\n` +
+          `# Disputes\n\n${this.getIssueDisputeString(issue.issueDisputes)}\n`;
+      default:
+        return issue.createGithubIssueDescription();
+    }
+  }
+
+  private getIssueDisputeString(issueDisputes: IssueDispute[]): string {
+    let issueDisputeString = '';
+    for (const issueDispute of issueDisputes) {
+      issueDisputeString += issueDispute.toString();
+    }
+    return issueDisputeString;
+  }
+
+  deleteIssue(id: number): Observable<Issue> {
+    return this.githubService.closeIssue(id).pipe(
+      map((response: GithubIssue) => {
+        const deletedIssue = this.createIssueModel(response);
+        this.deleteFromLocalStore(deletedIssue);
+        return deletedIssue;
+      })
+    );
+  }
+
+  /**
+   * This function will update the issue's state of the application. This function needs to be called whenever a issue is deleted.
+   */
+  deleteFromLocalStore(issueToDelete: Issue) {
+    const { [issueToDelete.id]: issueToRemove, ...withoutIssueToRemove } = this.issues;
+    this.issues = withoutIssueToRemove;
+    this.issues$.next(Object.values(this.issues));
+  }
+
+  /**
+   * This function will update the issue's state of the application. This function needs to be called whenever a issue is added/updated.
+   */
+  updateLocalStore(issueToUpdate: Issue) {
+    this.issues = {
+      ...this.issues,
+      [issueToUpdate.id]: issueToUpdate,
+    };
+    this.issues$.next(Object.values(this.issues));
+  }
+
+  /**
+   * Check whether the issue has been responded in the phase 2/3.
+   */
+  hasTeamResponse(issueId: number): boolean {
+    return !!this.issues[issueId].teamResponse;
+  }
+
+  /**
+   * Obtain an observable containing an array of issues that are duplicates of the parentIssue.
+   */
+  getDuplicateIssuesFor(parentIssue: Issue): Observable<Issue[]> {
+    return this.issues$.pipe(map((issues) => {
+      return issues.filter(issue => {
+        return issue.duplicateOf === parentIssue.id;
+      });
+    }));
+  }
+
+  reset() {
+    this.issues = undefined;
+    this.sessionId = undefined;
+    this.issues$.next(new Array<Issue>());
+
+    this.stopPollIssues();
+    this.isLoading.complete();
+    this.isLoading = new BehaviorSubject<boolean>(false);
+  }
+
+  /**
+   * Populates store with random issues depending on the current test phase.
+   */
+  private initializeData(): Observable<Issue[]> {
+    if (this.issues != null) {
+      return of(Object.values(this.issues));
+    }
+
+    const NUM_ISSUES = 10;
+    const generatedIssues: Array<GithubIssue> = [];
+
+    switch (this.phaseService.currentPhase) {
+      case Phase.phaseBugReporting:
+        for (let i = 0; i < NUM_ISSUES; i++) {
+          generatedIssues.push(generateIssueWithRandomData());
```

Comment:
I'm not very sure if our issues should be randomly generated.
We may want to test the "search bar" functionality, for which the issues need to be deterministic.
However, we can always create our own issues within the test.

Let's keep the current issue generation for now.

Reply:
Yes thats true. I thought the randomization would allow for a greater "variety" of issue properties so it could help weed out specific problems if there are any. Perhaps we could combine both a random set as well as a number of predetermined ones for when we need to filter specific issues like with the Search Bar. 👍